.github/workflows/create-release.yml

```diff
@@ -1,279 +1,456 @@
+
     steps:
-    - name: Run if stack files changed
-      run: |
-        echo "stack files have changed"
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Fetch previous build sha256 image hash code
+        id: fetch_previous_build_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_BUILD_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_BUILD_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Fetch previous run sha256 image hash code
+        id: fetch_previous_run_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_RUN_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_RUN_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Fetch previous run Node.js 16 sha256 image hash code
+        id: fetch_previous_run_nodejs_16_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_RUN_NODEJS_16_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_RUN_NODEJS_16_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Fetch previous run Node.js 18 sha256 image hash code
+        id: fetch_previous_run_nodejs_18_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_RUN_NODEJS_18_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_RUN_NODEJS_18_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Compare previous and current sha256 hash codes
+        id: compare_previous_and_current_sha256_hash_codes
+        run: |
+
+          ubi_images_need_update=()
+
+          # Run Image 
+          docker pull registry.access.redhat.com/ubi8/ubi-minimal
+          current_run_sha256_image_hash_code=$(docker image inspect registry.access.redhat.com/ubi8/ubi-minimal --format '{{json .RepoDigests}}' | jq -r '.[0]')
+          previous_run_sha256_image_hash_code=$(cat ${{ github.workspace }}/${{ env.UBI_RUN_SHA256_FILENAME }})
+
+          if [ "$previous_run_sha256_image_hash_code" != "$current_run_sha256_image_hash_code" ]; then
+            ubi_images_need_update+=("node-run")
+          fi
+
+          # Build Image 
```

Comment:
We also need to check:

registry.access.redhat.com/ubi8/nodejs-16
registry.access.redhat.com/ubi8/nodejs-18

as those will be updated if what we dnf install in the extension during the build phase (compilers,etc.) need updates. That set may be larger than what is in the minimal images.

Reply:
I might be missing something here in terms of context, but we dont use these images `registry.access.redhat.com/ubi8/nodejs-16 registry.access.redhat.com/ubi8/nodejs-18` on the current stack.
Also, do we have to change the content of the `stack/build.dockerfile` from this `FROM registry.access.redhat.com/ubi8/ubi-minimal:latest` to this `FROM registry.access.redhat.com/ubi8/ubi:latest`

Reply:
@pacostas lets sync on this, I don't understand your comment about changing the stack/build.dockerfile FROM.  I believe it has to match what is in the stack and that needs to be registry.access.redhat.com/ubi8/ubi:latest as the stack will be used to build nodejs, Java, etc images

Reply:
ok after some more discussion with @pacostas I agree we don't need to check ubi8/nodejs-16 or ubi8/nodejs-18 since we don't use them directly and the extension will install the latest version of the dnf installed packages and pick up any updates. It's only the run images and ubi8/ubi-minimal we need to check.